platform/core.network/src/org/netbeans/core/network/proxy/pac/impl/NbPacScriptEvaluator.java

```diff
@@ -336,37 +330,59 @@ private PacScriptEngine getScriptEngine(String pacSource) throws PacParsingExcep
 
             // Do some minimal testing of the validity of the PAC Script.
             final PacJsEntryFunction jsMainFunction;
-            if (nashornJava8u40Available) {
-                jsMainFunction = testScriptEngine(engine, true);
-            } else {
-                jsMainFunction = testScriptEngine(engine, false);
-            }
+            jsMainFunction = testScriptEngine(engine, false);
             
             return new PacScriptEngine(engine, jsMainFunction);
         } catch (ScriptException ex) {
             throw new  PacParsingException(ex);
         }
     }
     
-    private boolean getNashornJava8u40Available() {
+    private boolean isNashornFactory(ScriptEngineFactory f) {
         try {
             Class<?> klass = Class.forName("jdk.nashorn.api.scripting.NashornScriptEngineFactory");
+            return klass.isInstance(f);
         } catch (ClassNotFoundException ex) {
             return false;
         }
-        return true;
     }
-    
-    private ScriptEngine getNashornJSScriptEngine() {
-        NashornScriptEngineFactory factory = new NashornScriptEngineFactory();
-        return factory.getScriptEngine(new ClassFilterPacHelpers());
+
+    private ScriptEngine secureEngineEngine(ScriptEngine e) {
+        try {
+            ScriptEngineFactory f = e.getFactory();
+            final Class<? extends ScriptEngineFactory> factoryClass = f.getClass();
+            final ClassLoader factoryClassLoader = factoryClass.getClassLoader();
+            Class<?> filterClass = Class.forName("jdk.nashorn.api.scripting.ClassFilter", true, factoryClassLoader);
+            Method createMethod = factoryClass.getMethod("getScriptEngine", filterClass);
+            Object filter = java.lang.reflect.Proxy.newProxyInstance(factoryClassLoader, new Class[] { filterClass }, (Object proxy, Method method, Object[] args) -> {
+                return false;
+            });
+            return (ScriptEngine) createMethod.invoke(f, filter);
```

Comment:
Not sure I understand the ClassFilter. It always returns `false` ? 
In any case, it may be an idea to have unit test which tests if the ClassFilter actually gets invoked as we believe it should.

Btw: Note what really long lines does to a GitHub review !  Not too good at it myself but this is one more reason to cap lines at 80 chars or something like that.

Reply:
Right, I'll prepare some unit tests to make sure the engines are safe to all known sandbox escape strategies.

Reply:
There is `pac-test-getclass.js` now.

Reply:
Another test added in f47f36c